test/src/test/java/hudson/bugs/JnlpAccessWithSecuredHudsonTest.java

```diff
@@ -86,7 +87,7 @@ protected Slave createNewJnlpSlave(String name) throws Exception {
     public void anonymousCanAlwaysLoadJARs() throws Exception {
         r.jenkins.setNodes(Collections.singletonList(createNewJnlpSlave("test")));
         JenkinsRule.WebClient wc = r.createWebClient();
-        HtmlPage p = wc.login("alice").goTo("computer/test/");
+        HtmlPage p = wc.withBasicApiToken(User.getById("alice", true)).goTo("computer/test/");
```

Comment:
Why does simply `withBasicApiToken("alice")` not work here? Does that user not exist until created here?

Reply:
=> `java.lang.AssertionError: The userId must correspond to an already created User` if I just put Alice's login. The user must be created before using the methd and is not yet created.